doc/guides/build/index.rst

```diff
@@ -0,0 +1,132 @@
+.. _build_overview:
+
+Build Overview
+##############
+
+The Zephyr build process can be divided into two main phases: a
+configuration phase (driven by *CMake*) and a build phase (driven by
+*Make* or *Ninja*). This document is describing the build using *Make*
+as example.
+
+Configuration Phase
+*******************
+
+The configuration phase begins when the user invokes *CMake*,
+specifying a source application directory and a board target.
+
+.. figure:: build-config-phase.png
+    :align: center
+    :alt: Zephyr's build configuration phase
+    :figclass: align-center
+    :width: 80%
+
+*CMake* begins by processing the *CMakeLists.txt* file in the application
+directory, which will refer to the *CMakeLists.txt* file in the Zephyr
+top-level directory, which in turn will refer to *CMakeLists.txt* files
+throughout the build tree (directly and indirectly). Its primary
+output is a set of *Makefile*  to drive the build process, but *CMake*
+scripts do some build processing of their own:
+
+**Device tree** Using *cpp*, device-tree specifications (*.dts/.dtsi*
```

Comment:
This would render better as a "dictionary list" with a slight tweak.  Also since we're an introductory document, it might be good to spell out tool names on first use:

```
Device tree 
   Using the C preprocessor, *cpp*, device-tree specifications (*.dts/.dtsi*
   files) are collected from the target’s architecture, SoC, board, and
   application directories and compiled with the Device Tree Compiler, *dtc*.
   Then the build tool (*scripts/dts*) converts *dtc* output into *.h* files
   for later consumption.
```